BenchPress/Generators/AzureDeploymentImporter.cs

```diff
@@ -164,27 +171,56 @@ string outputFolderPath
     /// }
     /// </code>
     /// </summary>
-    private static Dictionary<string, object> GetExtraProperties(JsonNode resource)
+    private static Dictionary<string, string> GetExtraProperties(JsonNode resource)
     {
-        var extraProperties = new Dictionary<string, object>();
-
-        var dependsOn = resource["dependsOn"]?[0]?.ToString().Trim().Split(",");
-        var resourceIds = dependsOn?[0].Trim('\'').Split("/").Skip(1).ToArray();
-        var resourceNames = dependsOn?.Skip(1).ToArray();
-        if (
-            resourceIds != null
-            && resourceNames != null
-            && resourceIds.Length == resourceNames.Length
-        )
+        var extraProperties = new Dictionary<string, string>();
+        var dependencies = (JsonArray?) resource[s_dependsOnKey];
+
+        if (dependencies != null)
         {
-            foreach (var resourceId in resourceIds)
+            foreach (var dependency in dependencies)
             {
-                extraProperties.Add(
-                    resourceId,
-                    resourceNames[Array.IndexOf(resourceIds, resourceId)]
-                );
+                if (dependency != null)
+                {
+                    // There will be only one Capture for the Group that is the entire list of parameters that are
+                    // passed to "[resourceId()]" as a single string. After the split, the first parameter in

```

Comment:
I don't think I understand what `"[resourceId()]"` is.

Do we have a parameter name we could reference instead of or in addition to?

Reply:
Gotcha.